src/syntax-tree/ast.ts

```diff
@@ -2546,7 +2546,7 @@ export class Module {
                         isValid = this.focusedNode.type === code.returns || this.focusedNode.type === DataType.Any
 
                         //assign operand types based on argument type for expressions being used as args
-                        if(!isValid && this.focusedNode.rootNode instanceof FunctionCallStmt && code instanceof BinaryOperatorExpr){
+                        if(!isValid && this.focusedNode instanceof TypedEmptyExpr && code instanceof BinaryOperatorExpr){
```

Comment:
Need to be careful with how it affects Any. It should be fine since as soon as the user inputs an operand, the expression's second operand as well as its return data type change to that of the first operand. So even if it starts as Any, it will be changed on usage.